src/ProjectTemplates/ComponentsWebAssembly.ProjectTemplates/content/ComponentsWebAssembly-CSharp/Client/Program.cs

```diff
@@ -3,9 +3,11 @@
 using System.Collections.Generic;
 using System.Threading.Tasks;
 using System.Text;
+#if (!NoAuth && Hosted)
```

Comment:
`!NoAuth && Hosted` and its opposite `!Hosted || NoAuth` shows up in a few places. I find it quite taxing to reason about all the conditions!

Might it be worth defining another computed template variable `HostedAuth` defined as `!NoAuth && Hosted` so these usages can be simplified?

Reply:
Yeah, but then you end up with a combinatorial number of flags for each individual case. I'd rather keep things as they are, but if you feel strongly about it I can file a separate issue and we can revisit.

Our templates are complex and hard, there's no easy way around it, sadly.

Reply:
If you think this particular combination (including its negation) doesn't appear in too many places then sure, fine to leave as-is.